File: modules/react-native-sample/android/src/main/java/com/sample/entity/EndlessQuote.kt
package com.sample.entity

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.bridge.WritableMap

data class EndlessQuote(
    override val key: String,
    val symbolName: String,
    val tradePrice: String,
    val tradePriceStatus: QuoteBaseStatus,
    val point: String,
    val pointStatus: QuoteBaseStatus,
    val percent: String,
    val percentStatus: QuoteBaseStatus,
    override val isFooter: Boolean,
    override val isHeader: Boolean
) : BaseEntity {
    fun toMap(): WritableMap {
        return Arguments.createMap().apply {
            putString("key", key)
            putString("symbolName", symbolName)
            putString("tradePrice", tradePrice)
            putString("tradePriceStatus", tradePriceStatus.toString())
            putString("point", point)
            putString("pointStatus", pointStatus.toString())
            putString("percent", percent)
            putString("percentStatus", percentStatus.toString())
        }
    }

    companion object {
        fun fromArray(array: ReadableArray): ArrayList<EndlessQuote> {
            val resultArray = arrayListOf<EndlessQuote>()
            for (i in 0 until array.size()) {
                val map = array.getMap(i)
                resultArray.add(fromMap(map))
            }
            return resultArray
        }

        fun fromMap(map: ReadableMap): EndlessQuote {
            val key = map.getString("key") ?: ""
            val symbolName = map.getString("symbolName") ?: ""
            val tradePrice = map.getString("tradePrice") ?: ""
            val tradePriceStatus =
                QuoteBaseStatus.fromString(map.getString("tradePriceStatus") ?: "")
            val point = map.getString("point") ?: ""
            val pointStatus = QuoteBaseStatus.fromString(map.getString("pointStatus") ?: "")
            val percent = map.getString("percent") ?: ""
            val percentStatus = QuoteBaseStatus.fromString(map.getString("percentStatus") ?: "")
            val isFooter = false
            val isHeader = false

            return EndlessQuote(
                key = key,
                symbolName = symbolName,
                tradePrice = tradePrice,
                tradePriceStatus = tradePriceStatus,
                point = point,
                pointStatus = pointStatus,
                percent = percent,
                percentStatus = percentStatus,
                isFooter = isFooter,
                isHeader = isHeader
            )
        }
    }
}

File: modules/react-native-sample/ios/Entity/EndlessQuote.swift
import Foundation

struct EndlessQuote {
    let key                  : String
    let symbolName           : String
    let tradePrice           : String
    let tradePriceStatus     : QuoteBaseStatus
    let pointStatus          : QuoteBaseStatus
    let percentStatus        : QuoteBaseStatus
    let point                : String
    let percent              : String
    

    
    static func dic2EndlessQuote(dic: NSDictionary) -> EndlessQuote {
        let key                 = dic["key"]                as? String  ?? ""
        let symbolName          = dic["symbolName"]         as? String  ?? ""
        let tradePrice          = dic["tradePrice"]         as? String  ?? ""
        let tradePriceStatus    = dic["tradePriceStatus"]   as? String  ?? ""
        let pointStatus         = dic["pointStatus"]        as? String  ?? ""
        let percentStatus       = dic["percentStatus"]      as? String  ?? ""
        let point               = dic["point"]              as? String  ?? ""
        let percent             = dic["percent"]            as? String  ?? ""
        
        return EndlessQuote(key: key,
                     symbolName         : symbolName,
                     tradePrice         : tradePrice,
                     tradePriceStatus   : QuoteBaseStatus.fromString(value: tradePriceStatus),
                     pointStatus        : QuoteBaseStatus.fromString(value: pointStatus),
                     percentStatus      : QuoteBaseStatus.fromString(value: percentStatus),
                     point              : point,
                     percent            : percent
        )
    }
    
    static func arrayDicToArrayEndlessQuote(arr: Array<NSDictionary>) -> Array<EndlessQuote> {
        return arr.map(EndlessQuote.dic2EndlessQuote)
    }
}

extension EndlessQuote {
    func toDic() -> [String: Any] {
        return [
            "key"               : key,
            "symbolName"        : symbolName,
            "tradePrice"        : tradePrice,
            "tradePriceStatus"  : tradePriceStatus.rawValue,
            "pointStatus"       : pointStatus.rawValue,
            "percentStatus"     : percentStatus.rawValue,
            "point"             : point,
            "percent"           : percent
        ]
    }
}
